shims/spark311/src/main/scala/org/apache/spark/sql/rapids/shims/spark311/GpuInMemoryTableScanExec.scala

```diff
@@ -26,90 +27,95 @@ import org.apache.spark.sql.catalyst.plans.QueryPlan
 import org.apache.spark.sql.catalyst.plans.physical.Partitioning
 import org.apache.spark.sql.execution.{LeafExecNode, SparkPlan}
 import org.apache.spark.sql.execution.columnar.InMemoryRelation
-import org.apache.spark.sql.execution.metric.SQLMetrics
 import org.apache.spark.sql.vectorized.ColumnarBatch
 
-case class GpuInMemoryTableScanExec(
-  attributes: Seq[Attribute],
-  predicates: Seq[Expression],
-  @transient relation: InMemoryRelation) extends LeafExecNode with GpuExec {
-
-    override val nodeName: String = {
-      relation.cacheBuilder.tableName match {
-        case Some(_) =>
-          "Scan " + relation.cacheBuilder.cachedName
-        case _ =>
-          super.nodeName
-      }
+case class GpuInMemoryTableScanExec(attributes: Seq[Attribute],
+   predicates: Seq[Expression],
+   @transient relation: InMemoryRelation) extends LeafExecNode with GpuExec {
+
+  override val nodeName: String = {
+    relation.cacheBuilder.tableName match {
+      case Some(_) =>
+        "Scan " + relation.cacheBuilder.cachedName
+      case _ =>
+        super.nodeName
     }
+  }
+
+  override def innerChildren: Seq[QueryPlan[_]] = Seq(relation) ++ super.innerChildren
+
+  override def doCanonicalize(): SparkPlan =
+    copy(attributes = attributes.map(QueryPlan.normalizeExpressions(_, relation.output)),
+      predicates = predicates.map(QueryPlan.normalizeExpressions(_, relation.output)),
+      relation = relation.canonicalized.asInstanceOf[InMemoryRelation])
 
-    override def innerChildren: Seq[QueryPlan[_]] = Seq(relation) ++ super.innerChildren
-
-    override def doCanonicalize(): SparkPlan =
-      copy(attributes = attributes.map(QueryPlan.normalizeExpressions(_, relation.output)),
-        predicates = predicates.map(QueryPlan.normalizeExpressions(_, relation.output)),
-        relation = relation.canonicalized.asInstanceOf[InMemoryRelation])
-
-    override def vectorTypes: Option[Seq[String]] =
-      relation.cacheBuilder.serializer.vectorTypes(attributes, conf)
-
-    private lazy val columnarInputRDD: RDD[ColumnarBatch] = {
-      val numOutputRows = longMetric("numOutputRows")
-      val buffers = filteredCachedBatches()
-      relation.cacheBuilder.serializer.asInstanceOf[ParquetCachedBatchSerializer]
-        .gpuConvertCachedBatchToColumnarBatch(
-          buffers,
-          relation.output,
-          attributes,
-          conf).map { cb =>
+  override def vectorTypes: Option[Seq[String]] =
+    relation.cacheBuilder.serializer.vectorTypes(attributes, conf)
+
+  private lazy val columnarInputRDD: RDD[ColumnarBatch] = {
+    val rapidsConf = new RapidsConf(conf)
+    val numOutputRows = if (MetricsLevel(rapidsConf.metricsLevel) == DEBUG_LEVEL) {
+      longMetric("numOutputRows")
+    } else {
+      null
+    }
```

Comment:
Is this code necessary?  `GpuExec` is already conditionally creating the output rows metric at debug level, see https://github.com/NVIDIA/spark-rapids/blob/branch-0.5/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExec.scala#L217-L221.  I think this just needs to use the already created `GpuMetric` directly and not check for null later:
```suggestion
    val numOutputRows =gpuLongMetric(GpuMetrics.NUM_OUTPUT_ROWS)
```